.serena/memories/architecture.md

```
@@ -0,0 +1,179 @@
+$config = new SessionConfig(
+    new RedisConnectionConfig(),
+    new PhpSerializeSerializer(),
+    new DefaultSessionIdGenerator(),
+    (int)ini_get('session.gc_maxlifetime'),
+    new NullLogger()
+);
+$factory = new SessionHandlerFactory($config);
+$handler = $factory->build();
+```
+
+### ビルダーパターン
```

markdownlint-cli2 (0.18.1)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)